### ISSUE TITLE
Pass explict filenames, not subsets, for training

### ISSUE THREAD

**Issue by JimCircadian** [icenet-ai/icenet#263]
* IceNet version: 0.2.9

### Description

Running an `icenet_train` command, you don't pass the filename but rather the identifier for the dataset. This seems counter intuitive and I'm not entirely sure what it offers, ideally we should pass a filename. Thoughts @bnubald? 

### What I Did

```
$ icenet_train_tensorflow -nw -e 1 dataset_config.miniscule_bas_south.json local_test1 42

  File "/home/jambyr/icenet2/icenet/icenet/data/dataset.py", line 77, in __init__
    self._load_configuration(configuration_path)
  File "/home/jambyr/icenet2/icenet/icenet/data/dataset.py", line 130, in _load_configuration
    raise OSError("{} not found".format(path))
OSError: dataset_config.dataset_config.miniscule_bas_south.json.json not found
```

Note I'm using a dev branch so line numbers are awry, but this applies to the 0.2 release branches. We should change this and the equivalent of MergedIceNetDataSet

```
dataset = IceNetDataSet("dataset_config.{}.json".format(args.dataset),
                                batch_size=args.batch_size,
                                shuffling=args.shuffle_train)
```

**Comment by bnubald** [comment]
@JimCircadian, yes, is counter-intuitive, but assume would like to leave legacy option?

E.g. detect if file extension is json, and if it isn't, use existing approach, else, use filename.

**Comment by JimCircadian** [comment]
I like your style @bnubald, I'll stick this in to my existing work stream 😄

**Comment by bnubald** [comment]
Since resolved by JimCircadian in 5f56a1f but in `v0.3.0_dev` branch, I might cherry-pick relevant commit for either `v0.2.9_dev` or `v0.2.10_dev` since its quite helpful having the full filename.
